cstrml/market/src/lib.rs

```diff
@@ -859,9 +872,13 @@ impl<T: Config> Module<T> {
         
         // 1. Delete files anchor
         <Files<T>>::mutate(cid, |maybe_f| match *maybe_f {
-            Some((_, ref mut used_info)) => {
-                if used_info.groups.take(anchor).is_some() {
-                    used_size = used_info.used_size;
+            Some((ref file_info, ref mut used_info)) => {
+                if let Some(is_calculated_as_reported_group_count) = used_info.groups.remove(anchor) {
+                    used_size = used_info.used_size; // need to delete the used_size before the update
```

Comment:
what if `is_calculated_as_reported_group_count ` = false?

Reply:
we should still delete used no matter the value of `is_calculated_as_reported_group_count`. I added some comments here.